### DISCUSSION TITLE
How to use this on Android NDK?

### DISCUSSION THREAD

**Discussion by mavavilj** [mackron/dr_libs#242] [Q&A]
I've been trying to use this library on Android NDK, but I'm struggling with drwav_init_file failing even when I provide it a valid path.

Some example code for this would be useful.

BTW, my code works perfectly on my desktop (just change the path). But when I change the path to one got through JNI (/storage/emulated/0/Music/test.wav), then drwav_init_file fails as if the file path is invalid.

**Comment by mavavilj** [comment]
BTW, when printing the errno given by drwav_init_file it says:

socket operation on non-socket

Even when I see nothing about sockets.

OTOH, fopen on the same filepath seems to work.

**Comment by mackron** [comment]
dr_wav uses `fopen()`. I don't know what the rules are with `fopen()` and Android and when you can and can't use it - Android is not a platform I do frequent development on. You can do a search for `drwav_result drwav_fopen` to see what dr_wav is doing. It should be hitting one of the branches in the screenshot below. Maybe put a breakpoint on there to see how it's getting called.

![image](https://user-images.githubusercontent.com/1232553/189548894-6f7cdda4-368b-4a87-9b05-9fe9eac2cc5e.png)

**Comment by mavavilj** [comment]
Yes, and upon a closer inspection, I am not sure if this sort of library is even relevant to Android, because some sources suggest reading data in Java, not NDK. Though, libsndfile is for Android: https://github.com/libsndfile/libsndfile/blob/master/Building-for-Android.md

**Reply by mackron** [reply]
In my audio library miniaudio, in my tests I load the file data from Java and then pass a pointer to that data to native code and then just load from memory rather than a file.
